esp/services/ws_fs/ws_fsService.cpp

```diff
@@ -2916,28 +2916,36 @@ bool CFileSprayEx::onFileList(IEspContext &context, IEspFileListRequest &req, IE
             StringBuffer dzName;
             if (isEmptyString(dropZoneName))
                 dropZoneName = findDropZonePlaneName(netaddr, sPath, dzName);
-            if (!isEmptyString(dropZoneName))
+            Owned<IPropertyTree> dropZone;
+            if (!isEmptyString(dropZoneName)) // must be true, unless bare-metal and isDropZoneRestrictionEnabled()==false
+            {
+                dropZone.setown(getDropZonePlane(dropZoneName));
+                if (!dropZone)
+                    throw makeStringExceptionV(ECLWATCH_INVALID_INPUT, "Unknown landing zone: %s", dropZoneName);
+                if (!isPathInPlane(dropZone, sPath))
+                    throw makeStringExceptionV(ECLWATCH_INVALID_INPUT, "Path '%s' is not valid for dropzone '%s'", sPath.str(), dropZoneName);
+
                 checkDZPathScopePermissions(context, dropZoneName, sPath, SecAccess_Read);
+            }
 
             StringArray hosts;
-            if (isEmptyString(netaddr))
+            if (!isEmptyString(netaddr))
             {
-                Owned<IPropertyTree> dropZone = getDropZonePlane(dropZoneName);
-                if (!dropZone)
-                    throw makeStringExceptionV(ECLWATCH_INVALID_INPUT, "Unknown landing zone: %s", dropZoneName);
+                if (dropZone && !isHostInPlane(dropZone, netaddr, false))
```

Comment:
>Also, now you've consolidated the checks into checkDropZoneInputAndAccess, it looks like this code is very similar and checkDropZoneInputAndAccess should be used here too.

ditto. I think a call to checkDropZoneInputAndAccess should be used to validate in a common way.